lib/routes.py

```diff
@@ -23,16 +23,38 @@ def send_message(data):
 
 # Find a group
 def find_group(access_code):
-  Group.objects(access_code=access_code).get_or_404()
+  if Group.objects(access_code= f"{access_code}"):
+    return True
+  else:
+    return False
 
 @app.route('/groups', methods=['GET'])
 def get_groups():
   groups = Group.objects().to_json()
   return Response(groups, mimetype="application/json", status=200)
-  
+
 # # Join a group
-# @socketio.on('join_group')
-# def on_join_group(data):
+@socketio.on('join_group')
+def group_join(data):
+  if find_group(data['access_code']):
+    room = data['access_code']
+    try:
+      Connection(group=room, sid=request.sid).save()
+      join_room(room)
+    except:
+      print('something went wrong with adding the connection')
+    send('Successfully Connected to Group', room=room)
+  else:
+    raise ConnectionRefusedError('Group Not Found')
```

Comment:
I’m curious on how this will get sent to the front end - something we can test when a connection is tested